Diagnostics/FreeBusyChecker/FreeBusyChecker.ps1

```diff
@@ -0,0 +1,326 @@
+.PARAMETER OnPremDomain
+Specifies the domain for on-premises Organization.
+.PARAMETER OnPremEWSUrl
+Specifies the EWS (Exchange Web Services) URL for on-premises Exchange Server.
+.PARAMETER OnPremLocalDomain
+Specifies the local AD domain for the on-premises Organization.
+.PARAMETER Help
+Show help for this script.
+
+.EXAMPLE
+.\FreeBusyChecker.ps1
+This cmdlet will run the Free Busy Checker script and Check Availability OAuth and DAuth Configurations both for Exchange On-Premises and Exchange Online.
+.EXAMPLE
+.\FreeBusyChecker.ps1 -Auth OAuth
+This cmdlet will run the Free Busy Checker Script against OAuth Availability Configurations only.
+.EXAMPLE
+.\FreeBusyChecker.ps1 -Auth DAuth
+This cmdlet will run the Free Busy Checker Script against DAuth Availability Configurations only.
+.EXAMPLE
+.\FreeBusyChecker.ps1 -Org ExchangeOnline
+This cmdlet will run the Free Busy Checker Script for Exchange Online Availability Configurations only.
+.EXAMPLE
+.\FreeBusyChecker.ps1 -Org ExchangeOnPremise
+This cmdlet will run the Free Busy Checker Script for Exchange On-Premises OAuth and DAuth Availability Configurations only.
+.EXAMPLE
+.\FreeBusyChecker.ps1 -Org ExchangeOnPremise -Auth OAuth -Pause
+This cmdlet will run the Free Busy Checker Script for Exchange On-Premises Availability OAuth Configurations, pausing after each test is done.
+#>
+
+# Exchange On-Premises
+#>
+#region Properties and Parameters
+
+[Diagnostics.CodeAnalysis.SuppressMessageAttribute('PSUseDeclaredVarsMoreThanAssignments', '', Justification = 'Variables are being used')]
+[CmdletBinding(DefaultParameterSetName = "FreeBusyInfo_OP", SupportsShouldProcess)]
+
+param(
+    [Parameter(Mandatory = $false, ParameterSetName = "Test")]
+    [ValidateSet('DAuth', 'OAuth', 'All', '')]
+    [string[]]$Auth,
+    [Parameter(Mandatory = $false, ParameterSetName = "Test")]
+    [ValidateSet('ExchangeOnPremise', 'ExchangeOnline')]
+    [string[]]$Org,
+    [Parameter(Mandatory = $true, ParameterSetName = "Help")]
+    [switch]$Help,
+    [Parameter(Mandatory = $false, ParameterSetName = "Test")]
+    [string]$OnPremisesUser,
+    [Parameter(Mandatory = $false, ParameterSetName = "Test")]
+    [string]$OnlineUser,
+    [Parameter(Mandatory = $false, ParameterSetName = "Test")]
+    [string]$OnPremDomain,
+    [Parameter(Mandatory = $false, ParameterSetName = "Test")]
+    [string]$OnPremEWSUrl,
+    [Parameter(Mandatory = $false, ParameterSetName = "Test")]
+    [string]$OnPremLocalDomain
+)
+begin {
+    . $PSScriptRoot\Functions\OnPremDAuthFunctions.ps1
+    . $PSScriptRoot\Functions\OnPremOAuthFunctions.ps1
+    . $PSScriptRoot\Functions\ExoDAuthFunctions.ps1
+    . $PSScriptRoot\Functions\ExoOAuthFunctions.ps1
+    . $PSScriptRoot\Functions\htmlContent.ps1
+    . $PSScriptRoot\Functions\hostOutput.ps1
+} end {
+    function InstallRequiredModules {
+        try {
+            Get-Command -Module ActiveDirectory -ErrorAction Stop >$null
```

Comment:
We should be using `#Requires -Module ActiveDirectory` in the script inside of doing this. 

https://powershellisfun.com/2023/04/24/using-the-requires-statement-in-powershell/

Reply:
Adding #Requires -Module ActiveDirectory and removing this check

Reply:
Removed this funciton and function call as it is not needed

function InstallRequiredModules {
        try {
            Get-Command -Module ActiveDirectory -ErrorAction Stop >$null
        } catch {
            Import-Module ActiveDirectory
        }

        if (-not (Get-Module -Name ExchangeOnlineManagement -ListAvailable)) {
            Disconnect-ExchangeOnline -Confirm:$False
        }
    }